server/ERNI.PBA.Server.Host/Controllers/BudgetController.cs

```diff
@@ -76,24 +54,36 @@ public async Task<IActionResult> GetUserBudgetByYear(int budgetId, CancellationT
         [HttpGet("user/current/year/{year}")]
         public async Task<IActionResult> GetCurrentUserBudgetByYear(int year, CancellationToken cancellationToken)
         {
-            var budgets = await _budgetRepository.GetBudgets(HttpContext.User.GetId(), year, cancellationToken);
-
-            var result = budgets.Select(budget => new
+            var userId = HttpContext.User.GetId();
+            var budgets = await _budgetRepository.GetSingleBudgets(userId, year, cancellationToken);
+            var result = budgets.Select(budget =>
             {
-                Id = budget.Id,
-                Year = budget.Year,
-                Amount = budget.Amount,
-                AmountLeft = budget.Amount - budget.Requests.Sum(_ => _.Amount),
-                Title = budget.Title,
-                Type = budget.BudgetType,
-                Requests = budget.Requests.Select(_ => new
+                var transactions = budget.Requests.SelectMany(x => x.Transactions).ToList();
+                var amount = budget.Amount;
+                var amountLeft = amount - transactions.Sum(x => x.Amount);
+
+                return new
                 {
-                    Id = _.Id,
-                    Title = _.Title,
-                    Amount = _.Amount,
-                    Date = _.Date,
-                    State = _.State,
-                })
+                    Id = budget.Id,
+                    Year = budget.Year,
+                    Amount = amount,
+                    AmountLeft = amountLeft,
+                    Title = budget.Title,
+                    Type = budget.BudgetType,
+                    Requests = budget.Requests.Select(_ => new
+                    {
+                        Id = _.Id,
+                        Title = _.Title,
+                        Amount = _.Amount,
+                        Date = _.Date,
+                        State = _.State,
+                        Transactions = _.Transactions.Select(x => new
```

Comment:
Does the client need to know anything about transactions? (Especially since we are handling the team budgets separately now)